buildAndReleaseTask/pulumi.ts

```diff
@@ -102,26 +103,37 @@ export async function runPulumi() {
         tl.debug(tl.loc("Debug_PrintingVersion"));
         await tl.exec(toolPath, "version");
 
+        const agentEnvVars = tryGetEnvVars();
+        const loginEnvVars = {
+            ...agentEnvVars,
+        };
+        
+        const azureStorageContainer = agentEnvVars["AZURE_STORAGE_CONTAINER"];
+
         // Login and then run the command.
         tl.debug(tl.loc("Debug_Login"));
         // For backward compatibility, also check for `pulumi.access.token`
         // and manually set the access token env var for the login command.
         const pulumiAccessToken =
             tl.getVariable("pulumi.access.token") ||
+            // `getVariable` will automatically prepend the SECRET_ prefix if it finds
+            // it in the build environment's secret vault.
             tl.getVariable("PULUMI_ACCESS_TOKEN");
-        if (!pulumiAccessToken) {
-            tl.setResult(tl.TaskResult.Failed, tl.loc("PulumiAccessTokenNotFound"));
+        if (!azureStorageContainer && !pulumiAccessToken) {
+            tl.setResult(tl.TaskResult.Failed, tl.loc("PulumiLoginUndetermined"));
             return;
         }
 
-        const agentEnvVars = tryGetEnvVars();
-        const loginEnvVars = {
-            ...agentEnvVars,
-        };
-        loginEnvVars[PULUMI_ACCESS_TOKEN] = pulumiAccessToken;
+        var loginCommand = ["login"];
+        if (azureStorageContainer) {
+            loginCommand = ["login","-c", "azblob://"+azureStorageContainer];
+        } else {
+            loginEnvVars[PULUMI_ACCESS_TOKEN] = pulumiAccessToken;
```

Comment:
So the PR build failed with an error on this line. 

> pulumi.ts(131,13): error TS2322: Type 'string | undefined' is not assignable to type 'string'.

Not sure if you are using VS Code, but if you were, you should have  gotten a TSLint error here for trying to use `pulumiAccessToken` without a check for `undefined`. Let's update lines 128 to 132 to:

```
if (azureStorageContainer) {
  loginCommand = ["login","-c", "azblob://"+azureStorageContainer];
}  else if (pulumiAccessToken) {
  loginEnvVars[PULUMI_ACCESS_TOKEN] = pulumiAccessToken;
} else {
  tl.setResult(tl.TaskResult.Failed, tl.loc("PulumiLoginUndetermined"));
  return;
}
```

You can then also remove the lines 122 to 125 since we sort of would be doing the same thing with the `else...`.

Reply:
Feel free to do the changes as you see fit, i think it will be faster that way.

Reply:
I added the change, but removed the last "else"-clause, since it is already covered in line 122 (I hope!)